workspaces/tech-radar/plugins/tech-radar/src/defaultApi.ts

```diff
@@ -223,8 +224,76 @@ export const mock: TechRadarLoaderResponse = {
   rings,
 };
 
-export class SampleTechRadarApi implements TechRadarApi {
+export type Options = {
+  /**
+   * URL provided via config to load the graph data from.
+   */
+  url?: string;
+  /**
+   * Hard coded graph data populated directly via the config.
+   */
+  graphData?: Config;
+};
+
+export class DefaultTechRadarApi implements TechRadarApi {
+  private readonly url: string | undefined;
+  private readonly graphData: Config | undefined;
+
+  constructor(opts: Options) {
+    this.url = opts.url;
+    this.graphData = opts.graphData;
+  }
   async load() {
+    if (this.url) {
+      const response = await fetch(this.url);
+
+      if (!response.ok) {
+        throw new Error(
+          `Get request failed to ${this.url} with ${response.status} ${response.statusText}`,
+        );
+      }
+      const responseJson = await response.json();
+      if (!responseJson.path.endsWith('.json')) {
+        throw new Error(
+          `Retrieved file content from ${this.url} is not JSON. Please provide a URL to a JSON file.`,
+        );
+      }
+      const jsonContent: TechRadarLoaderResponse = JSON.parse(
+        atob(responseJson.content),
+      );
+      // could add more robust type validation here as a future improvement
```

Comment:
Waffled about writing a validator (or using zod/ts-interface-checker) to put a more robust type check on the JSON we get back from the URL here, but decided against given malformatted JSON will fail fast and can be easily corrected. Can implement if reviewers feel differently though!

Reply:
good point, let's keep zod as an option once we have decided the direction we want to take